### ISSUE TITLE
Metadata validation

### ISSUE THREAD

**Issue by chartgerink** [p2pcommons/sdk-js#76]
I expect the `sdk-js` to handle the metadata and validate it for corruptions of the specification. The end result should be that using the `sdk-js` makes it extremely difficult to be out-of-spec with @p2pcommons/specs

Currently, I can mess around with the metadata quite easily, which is especially important to be validated prior to publishing (previously called register). For example, I manually edited `dat.json` to be invalid (e.g., titles longer than 300 chars), but I could still publish to profile.

In other words, we'd need to validate the metadata prior to handling it at any timepoint. We could introduce a generic validator method to implement the specs and check the metadata at various timepoints. This would also give us one point of entry if the specs change and it would propagate to all the validations.

**Comment by juliangruber** [comment]
From an implementation point of view we can add this validation logic to the function that reads the metadata, so whenever it is read it is validated.

**Comment by dpaez** [comment]
Cool. Thanks for the feedback and opening the issue @chartgerink. 

There are some validation mechanisms in place (see [schemas](https://github.com/p2pcommons/sdk-js/tree/master/schemas)) but they are not up to date to recent changes like the 300 char limit on the title. 

I'll make use of this issue to track and update those that are missing.

**Comment by jameslibscie** [comment]
@dpaez @chartgerink Is it clear what needs validating, how it should be validated, when validation should occur? And to what extent is this implemented now?

**Comment by dpaez** [comment]
Hi @jameslibscie, I left this open as a flag for keeping an eye on every possible metadata validation issue, but I'm thinking that won't be pretty much effective. It would be ideal to have like many small issues with the specific problem that a big one. 
So for example, this issue describes one open problem, the 300 max limit, I need to add those to the schemas.

**Comment by jameslibscie** [comment]
Created a new issue for the 300 max limit (#116)
